Task-56521: Fix duplication of news and activity items in favorites drawer

SaraBoutej opened the pull request:
Before this Fix when we bookmark a news, we have the related activity that is bookmarked too, so in the drawer of recent favorites we have duplication of items: activity + news.
With this fix, we add a boolean property "hidden" when bookmark an item to avoid the duplication of an item in the favorites drawer.

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

SaraBoutej committed:
Task-56521: Fix duplication of news and activity items in favorites drawer

Amine0008 committed:
Task-56509: Configure News Application as a main page application #508

This change will allow to capture the main application name. In fact, any application will be able to define the variable eXo.env.portal.mainApplicationName (with the application/module name) inside its required JS in order to add an additional data property inside collected operation pageFullUIDisplay

Amine0008 committed:
Task-56509: Configure News Application as a main page application #508

This change will allow to capture the main application name. In fact, any application will be able to define the variable eXo.env.portal.mainApplicationName (with the application/module name) inside its required JS in order to add an additional data property inside collected operation pageFullUIDisplay

montassarhamadi committed:
Task 56744: Search by favorites and news (article bookmarked/unbookmarked from the activity ) (#509)

Problem: news still displays in search by using favorite filtering after unbookmarking.
Fix: the process of deleting activity was completed but not the case for activity details (news) because there is a wrong comparison which stopped the deleting of activity details. that's why the details stay saved and then displayed in search and also in recent favorites.

azayati committed:
Task-55805: [Accessibility - Contrast - 3.2] Change the default primary color (#517)

azayati committed:
Task-55805: [Accessibility - Contrast - 3.2] Change the default primary color (#517)

boubaker commented on webapp/src/main/webapp/news-details/components/ExoNewsFavoriteAction.vue:
```diff
@@ -65,7 +66,7 @@ export default {
     added() {
       this.displayAlert(this.$t('Favorite.tooltip.SuccessfullyAddedAsFavorite', {0: this.$t('news.label')}));
-      this.$favoriteService.addFavorite('news', this.news.id)
+      this.$favoriteService.addFavorite('news', this.news.id, '', true)
```
```suggestion
      this.$favoriteService.addFavorite('news', this.news.id, null, true)
```

boubaker commented on webapp/src/main/webapp/news-details/components/ExoNewsFavoriteAction.vue:
```diff
@@ -2,6 +2,7 @@
   <favorite-button
     :id="activityId"
     :favorite="isFavorite"
+    :hidden=true
```
```suggestion
    hidden
```
You can just add the name of the attribute to change its value to `true`
See https://v2.vuejs.org/v2/guide/components-props.html#Passing-a-Boolean

lbengayed committed:
Task-56827: Add text truncate in mobile  when space name is too long (#522)

hakermi committed:
TASK-56990: Wrong last modifier value  in news drafts management (#525)

Prior to this change, when a user (redactor) creates a news article in a space and then modify its content without saving to keep a draft, at this case when another redactor edits the article a warning message appears
to tell him that he is about to edit a drfat created by user 1, but instead it says that it's a draft created by user x which is the last user reader and not editor. this due to the fact that the exo:lastModifier is updating when add , remove or update a property in the jcr node which is the case when is the case when mark a news as read, so the username of the reader will be set in th exo:lastModifier value.
This PR adds a new mixin called exo:newsModification with an exo:newsLastModifier property, which will be updated when update the news and not when marking it as read.

houssemexo26 committed:
Merge branch 'feature/maintenance' into Task-56521
